### ISSUE TITLE
(aws_eks): (COMPUTE7_GRAVITON3 picks wrong AMI)

### ISSUE THREAD

**Issue by dborysenko** [aws/aws-cdk#20758]
### Describe the bug

Below code picks X86_64 AMI for ARM AutoScaling group.
```python
cluster.add_auto_scaling_group_capacity(
        "dmzASG",
        instance_type=ec2.InstanceType.of(ec2.InstanceClass.COMPUTE7_GRAVITON3,
                                          ec2.InstanceSize.XLARGE8),
        block_devices=[
            autoscaling.BlockDevice(device_name='/dev/xvda',
                                    volume=autoscaling.BlockDeviceVolume.ebs(
                                        50, encrypted=True))
        ],
        min_capacity=1,
        max_capacity=100,
        vpc_subnets=ec2.SubnetSelection(subnets=dmz_subnet_list)
)
```

### Expected Behavior

Expect 1 EC2 instance of type "c7g.8xlarge" to be created.

### Current Behavior

0 instances created. AutoScailing group Activity tab reports below error:
`
Launching a new EC2 instance. Status Reason: The architecture 'arm64' of the specified instance type does not match the architecture 'x86_64' of the specified AMI. Specify an instance type and an AMI that have matching architectures, and try again. You can use 'describe-instance-types' or 'describe-images' to discover the architecture of the instance type or AMI. Launching EC2 instance failed.
`

### Reproduction Steps

try to add asg capacity to eks cluster:
```python
cluster.add_auto_scaling_group_capacity(
        "dmzASG",
        instance_type=ec2.InstanceType.of(ec2.InstanceClass.COMPUTE7_GRAVITON3,
                                          ec2.InstanceSize.XLARGE8),
        block_devices=[
            autoscaling.BlockDevice(device_name='/dev/xvda',
                                    volume=autoscaling.BlockDeviceVolume.ebs(
                                        50, encrypted=True))
        ],
        min_capacity=1,
        max_capacity=100,
        vpc_subnets=ec2.SubnetSelection(subnets=dmz_subnet_list)
)
```

### Possible Solution

Switching InstanceClass to COMPUTE6_GRAVITON2 "resolves" the issue:
```python
cluster.add_auto_scaling_group_capacity(
        "dmzASG",
        instance_type=ec2.InstanceType.of(ec2.InstanceClass. COMPUTE6_GRAVITON2,
                                          ec2.InstanceSize.XLARGE8),
        block_devices=[
            autoscaling.BlockDevice(device_name='/dev/xvda',
                                    volume=autoscaling.BlockDeviceVolume.ebs(
                                        50, encrypted=True))
        ],
        min_capacity=1,
        max_capacity=100,
        vpc_subnets=ec2.SubnetSelection(subnets=dmz_subnet_list)
)
```

### CDK CLI Version

2.28.0 (build ba233f0)

### Node.js Version

v16.15.0

### OS

MacOS 12.4

### Language

Python

### Language Version

Python 3.9.12

**Comment by gergnz** [comment]
As soon as I can get my integration tests working this should be fixed in #20543

**Comment by peterwoodworth** [comment]
Thanks for getting to this @gergnz!
